File: 102. Binary Tree Level Order Traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(!root) return { };
        vector<vector<int>> res;
        queue<TreeNode*> q{{root}};
        //q.push(root);

        while(!q.empty()){
            vector<int> temp;
            int n = q.size();

            while(n){
                TreeNode *head = q.front();
                q.pop();
                temp.push_back(head->val);
                if(head->left) q.push(head->left);
                if(head->right) q.push(head->right);
                n--;
            }
            res.push_back(temp);
        }
        return res;
    }
};

File: Data Structure/Tree/Binary Tree/102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if not root:
            return []
        result_levels = []

        dequenode = deque([root])

        while dequenode:
            length = len(dequenode)
            level = []
            for i in range(length):
                node = dequenode.popleft()
                level.append(node.val)
                if node.left:
                    dequenode.append(node.left)
                if node.right:
                    dequenode.append(node.right)
            result_levels.append(level)

        return result_levels
        
